test/checkImageSize.js

```diff
@@ -0,0 +1,87 @@
+var fixImage = require("./fixImageSize.js");
+const sizeOf = require('image-size');
+const fs = require('fs');
+const { exit } = require("process");
+
+function getFileSizeMB(fileStats){
+    return (fileStats.size) / (1024*1024);
+}
+
+//Check if image is larger than 1MB
+function isFileSizeBig(fileStats){ 
+    return getFileSizeMB(fileStats) > 1;
+}
+
+function getImageWidth(imageSize){
+    return imageSize.width;
+}
+
+function getImageHeight(imageSize){
+    return imageSize.height;
+}
+
+//Check if the width of image is larger than 1024px or lower than 512px
+function isImageWidthUnreasonable(imageSize){
+    let width = imageSize.width;
+    return (width <512 || width > 1024)
+}
+
+function imageSizeChecker(input){  //input here is the image file
+    //let inputName = "./img/" + input;
+    let imageSize = sizeOf(input);
+    let width = imageSize.width;
+    let height = imageSize.height;
+    let imageStats = fs.statSync(input);
+    if (width != height){
+        throw Error("Not a pixel-perfect square aspect ratio")
+    }
+    else{
+        if(isFileSizeBig(imageStats)){
+            throw Error("Your file is " + getFileSizeMB(imageStats).toFixed(2) + "MB")
+        } 
+        else if(isImageWidthUnreasonable(imageSize)){
+            throw Error("Your image size is " + getImageWidth(imageSize) +"px by " +getImageHeight(imageSize)+"px");
+        }
+    }
+};
+
+//Loop through every file in "./img"
+//If the file has a "." included, it means that it contains .jpg, .png, etc
+//Those files are appended to a new array of all the file names
+//If not, it is a directory inside a directory and the files inside there need to be extracted
+//Again files with a "." are appended to the new array, while the other files are directories where its files need to be extracted
+//This loop continues until there is no other directory within a directory
+function getAllFiles(files, newFileArray, directoryName){ 
+    for (file of files){
+        if (file.includes(".")){
+            newFileArray.push(directoryName + "/" + file)
+        }
+        else{
+            newFile = fs.readdirSync(directoryName + "/" + file)
+            let newDirectoryName = directoryName + "/" + file
+            getAllFiles(newFile, newFileArray, newDirectoryName)
+        }
```

Comment:
FWIW I don't think you need to deal with recursively exploring folders, the top-level folder only has images under the hierarchy. I think not doing anything / explicit `continue` is the best-case thing to do!